smartsim/_core/generation/generator.py

```diff
@@ -249,10 +271,56 @@ def _build_tagged_files(tagged: TaggedFilesHierarchy) -> None:
 
             # write in changes to configurations
             if isinstance(entity, Model):
-                logger.debug(
-                    f"Configuring model {entity.name} with params {entity.params}"
+                files_to_params = self._writer.configure_tagged_model_files(
+                    to_write, entity.params
                 )
-                self._writer.configure_tagged_model_files(to_write, entity.params)
+                self._log_params(entity, files_to_params)
+
+    def _log_params(
+        self, entity: Model, files_to_params: t.Dict[str, t.Dict[str, str]]
+    ) -> None:
+        """Log which files were modified during generation
+
+        and what values were set to the parameters
+
+        :param entity: the model being generated
+        :type entity: Model
+        :param files_to_params: a dict connecting each file to its parameter settings
+        :type files_to_params: t.Dict[str, t.Dict[str, str]]
+        """
+        used_params: t.Dict[str, str] = {}
+        file_to_tables: t.Dict[str, str] = {}
+        for file, params in files_to_params.items():
+            used_params.update(params)
+            table = tabulate([[param, value] for param, value in params.items()],
+                             headers=["Name", "Value"])
+            file_to_tables[relpath(file, self.gen_path)] = table
+
+        if used_params:
+            used_params_str = ", ".join(
+                [f"{name}={value}" for name, value in used_params.items()]
+            )
+            logger.log(
+                level=self.log_level,
+                msg=f"Configured model {entity.name} with params {used_params_str}",
+            )
+            file_table = tabulate(
+                    [[file, table] for file, table in file_to_tables.items()],
```

Comment:
Same cast from `dict_items` to `list` here. If you decide to change above, it might be worth changing here as well.

If you leave the above "as is", feel free to resolve this comment

Reply:
Done!